### ISSUE TITLE
exported_namespace, exported_pod instead of namespace, pod?

### ISSUE THREAD

**Issue by gytisgreitai** [coreos/prometheus-operator#356]
**What did you do?**

hack/cluster-monitoring/deploy

**What did you expect to see?**

Ability to select pods, namespaces, etc. in graphana

**What did you see instead? Under which circumstances?**

Only `monitoring` namespace but nothing more

**Environment**

* Kubernetes version information:

```
Client Version: version.Info{Major:"1", Minor:"6", GitVersion:"v1.6.2", GitCommit:"477efc3cbe6a7effca06bd1452fa356e2201e1ee", GitTreeState:"clean", BuildDate:"2017-04-19T20:33:11Z", GoVersion:"go1.7.5", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"6", GitVersion:"v1.6.2", GitCommit:"477efc3cbe6a7effca06bd1452fa356e2201e1ee", GitTreeState:"clean", BuildDate:"2017-04-19T20:22:08Z", GoVersion:"go1.7.5", Compiler:"gc", Platform:"linux/amd64"}
```

* Kubernetes cluster kind: 

	GKE

In the example `Pods` dashboard there is a template variable with query set to :
`label_values(kube_pod_info, namespace)` which in my case returns only `monitoring` namespace.

For some reason to be able to retrieve pods, namespaces, I must use `exported_namespace` instead of `namespace` and `exported_pod` instead of `pod`.

Sample output from `kube_deployment_status_replicas` prometheus:

```
kube_pod_info{created_by="ReplicaSet/heapster-v1.3.0-191291410",endpoint="http-metrics",exported_namespace="kube-system",exported_pod="heapster-v1.3.0-191291410-bz83k",host_ip="10.132.0.2",instance="10.84.5.16:8080",job="kube-state-metrics",namespace="monitoring",node="gke-dev-default-pool-efca0068-hb1z",pod="kube-state-metrics-1587421894-lb1g3",pod_app="kube-state-metrics",pod_ip="10.84.4.13",service="kube-state-metrics",svc_app="kube-state-metrics",svc_k8s_app="kube-state-metrics"}
```

What am I doing wrong?

**Comment by brancz** [comment]
This is due to a recent change that allows us to use the `honor_labels` configuration of Prometheus. So master of kube-prometheus actually reflects the use of master of the Prometheus Operator. We hope to soon have releases of kube-prometheus as helm charts, to be able to separately release it to ensure stable installable packages.

**Comment by gytisgreitai** [comment]
Thanks for explanation, seems clear now:)

Few side questions if I may:

in `kubernetes-pods-dashboard.json` there is a query which uses 
`kube_pod_container_requested_memory_bytes` imho this should be `kube_pod_container_resource_requests_memory_bytes`

Any way to query by custom pod labels? It seems that I need relabelling feature?

**Comment by brancz** [comment]
There was a discussion in kube-state-metrics and that's what was agreed on. Feel free to open an issue there if you feel it should be changed. I don't have a particularly strong opinion about it, just out of compatibility I would hesitate to change it, but before the 1.0 it would still be ok.

The latest release of kube-state-metrics has the `kube_pod_labels`, `kube_node_labels`, and `kube_service_labels` that you can join onto any other metric.

**Comment by gytisgreitai** [comment]
Probably a slight misunderstanding:

`kubernetes-pods-dashboard.json` uses `kube_pod_container_requested_memory_bytes`
`resource-requests-dashboard.json` uses `kube_pod_container_resource_requests_memory_bytes`

So imho `kubernetes-pods-dashboard.json` is buggy cause there is no such metric anymore accoriding to this commit:

https://github.com/kubernetes/kube-state-metrics/commit/54ad74dfda739f4502d8de037ec572a194778a96

But maybe I'm getting something wrong.

**Comment by brancz** [comment]
I see. Sorry that was a misunderstanding on my end, yes that dashboard needs to be updated to use `kube_pod_container_resource_requests_memory_bytes`.

**Comment by brancz** [comment]
Feel free to open a PR for that as well :)

**Comment by gytisgreitai** [comment]
Created https://github.com/coreos/prometheus-operator/pull/359

**Comment by brancz** [comment]
#359 is merged and v0.9.0 of the Prometheus Operator is released, so I'm closing this here. Thanks a lot for your contributions @gytisgreitai !
